sky/spot/spot_utils.py

```diff
@@ -440,15 +442,40 @@ def load_spot_job_queue(payload: str) -> List[Dict[str, Any]]:
     return jobs
 
 
+@typing.overload
```

Comment:
For some reason mypy throws
```
sky/spot/spot_utils.py:446: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
```
@Michaelvll any obvious mistakes here?

Reply:
It seems overload argument with default value, needs to be the first kwargs. The following should work
```

@typing.overload
def format_job_table(tasks: List[Dict[str, Any]],
                     show_all: bool,
                     *
                     return_rows: Literal[False] = False,
                     max_jobs: Optional[int] = None) -> str:
    ...

@typing.overload
def format_job_table(tasks: List[Dict[str, Any]],
                     show_all: bool,
                     *
                     return_rows: Literal[True],
                     max_jobs: Optional[int] = None) -> List[List[str]]:
    ...

def format_job_table(tasks: List[Dict[str, Any]],
                     show_all: bool,
                     *
                     return_rows: bool = False,
                     max_jobs: Optional[int] = None) -> Union[str, List[List[str]]]:
    ...
```

Reply:
Hmm, same error on my local.

Reply:
Could you try to remove the cache of mypy? It seems working on my side.

Reply:
Removed cache, still doesn't pass locally. Pushed some changes to this branch to see.

Reply:
Ahh, the `return_rows: Literal[True]` should not have the `= True` in the argument, as the default value of that argument is `False`.